tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/asm/OCLAssembler.java

```diff
@@ -74,6 +74,8 @@ public OCLAssembler(TargetDescription target) {
             emitLine("#pragma OPENCL EXTENSION cl_khr_fp64 : enable  ");
         }
 
+        emitLine("#pragma OPENCL EXTENSION cl_khr_fp16 : enable  ");
```

Comment:
so, now we are going to emit this in all opencl kernels we generate, regardless if it uses half floating point?

Reply:
I think we should emit this line only if the platform supports it.